src/Stratis.Bitcoin.Features.Consensus/Behaviors/ProvenHeadersConsensusManagerBehavior.cs

```diff
@@ -214,6 +236,18 @@ protected override GetHeadersPayload BuildGetHeadersPayload()
             }
         }
 
+        /// <inheritdoc />
+        protected override Task ProcessHeadersAsync(INetworkPeer peer, List<BlockHeader> headers)
+        {
+            if (this.isGateway)
```

Comment:
If we are a gateway we just don't processes PH

Reply:
yes this change was to honor that